app.go

```diff
@@ -1,98 +1,180 @@
+	})
+	batchLimit := app.Int(cli.IntOpt{
+		Name:   "batchLimit",
+		Desc:   "The size of data in bytes, that a bulk write is writing into mongodb at once. Not recommended to use more than 16MB (e.g. 15000000)",
+		EnvVar: "BATCH_LIMIT",
+		Value:  15000000,
+	})
+
+	app.Command("scheduled-backup", "backup a set of mongodb collections", func(cmd *cli.Cmd) {
+		cronExpr := cmd.String(cli.StringOpt{
+			Name:   "cron",
+			Desc:   "Cron expression for when to run",
+			EnvVar: "CRON",
+			Value:  "30 10 * * *",
+			//Value:  "@every 30s",
+		})
+		dbPath := cmd.String(cli.StringOpt{
+			Name:   "dbPath",
+			Desc:   "Path to store boltdb file",
+			EnvVar: "DBPATH",
+			Value:  "/var/data/mongobackup/state.db",
+		})
+		run := cmd.Bool(cli.BoolOpt{
+			Name:   "run",
+			Desc:   "Run backups on startup?",
+			EnvVar: "RUN",
+			Value:  true,
+		})
+		healthHours := cmd.Int(cli.IntOpt{
+			Name:   "health-hours",
+			Desc:   "Number of hours back in time in which healthy backup needs to exist of each named collection for the app to be healthy. (e.g. 24)",
+			EnvVar: "HEALTH_HOURS",
+			Value:  77,
+		})
+
+		cmd.Action = func() {
+			parsedColls, err := parseCollections(*colls)
+			if err != nil {
+				log.Fatalf("error parsing collections parameter: %v\n", err)
+			}
+			dbService := newMongoService(*connStr, &labixMongo{}, &defaultBsonService{}, time.Duration(*mongoTimeout)*time.Second, time.Duration(*rateLimit)*time.Millisecond, *batchLimit)
+			statusKeeper, err := newBoltStatusKeeper(*dbPath)
+			if err != nil {
+				log.Fatalf("failed setting up to read or write scheduled backup status results: %v\n", err)
+			}
+			defer statusKeeper.Close()
+			storageService := newS3StorageService(*s3bucket, *s3dir, *s3domain, *accessKey, *secretKey)
+			backupService := newMongoBackupService(dbService, storageService, statusKeeper)
+			scheduler := newCronScheduler(backupService, statusKeeper)
+			healthService := newHealthService(*healthHours, statusKeeper, parsedColls, healthConfig{
+				appSystemCode: "up-mgz",
+				appName:       "mongobackup",
+			})
+			httpService := newScheduleHTTPService(scheduler, healthService)
+			httpService.ScheduleAndServe(parsedColls, *cronExpr, *run)
+		}
+	})
 
 	app.Command("backup", "backup a set of mongodb collections", func(cmd *cli.Cmd) {
-		colls := cmd.String(cli.StringArg{
-			Name:   "COLLECTIONS",
-			Desc:   "Collections to process (comma separated <database>/<collection>)",
-			EnvVar: "MONGODB_COLLECTIONS",
-			Value:  "foo/content,foo/bar",
+		dbPath := cmd.String(cli.StringOpt{
+			Name:   "dbPath",
+			Desc:   "Path to store boltdb file",
+			EnvVar: "DBPATH",
+			Value:  "/var/data/mongobackup/state.db",
 		})
+
 		cmd.Action = func() {
-			m := newMongolizer(*connStr, *s3bucket, *s3dir, *s3domain, *accessKey, *secretKey)
-			if err := m.backupAll(*colls); err != nil {
+			parsedColls, err := parseCollections(*colls)
+			if err != nil {
+				log.Fatalf("error parsing collections parameter: %v\n", err)
```

Comment:
Is it necessary to have a newline at the end of each log message?

Reply:
removed